testutil/fixtures/plugin/plugin.go

```diff
@@ -91,6 +91,7 @@ func CreateInRegistry(ctx context.Context, repo string, auth *types.AuthConfig,
 	}
 
 	var cfg Config
+	cfg.PluginConfig = &types.PluginConfig{}
```

Comment:
Thinking what would be better; set `PluginConfig` here, or update `TestPluginInstallArgs` to take this case into account; https://github.com/moby/moby/blob/07e6b843594e061f82baa5fa23c2ff7d536c2a05/integration-cli/docker_cli_plugins_test.go#L199-L201


And change the above to:

```go
	plugin.CreateInRegistry(ctx, pName, nil, func(cfg *plugin.Config) {
		if cfg.PluginConfig == nil {
			cfg.PluginConfig = &types.PluginConfig{}
		}
		cfg.Env = []types.PluginEnv{{Name: "DEBUG", Settable: []string{"value"}}}
	})
```

Reply:
@cpuguy83 @AkihiroSuda WDYT?

Reply:
how about this:
```golang
cfg := Config{PluginConfig: &types.PluginConfig{}}
```

Reply:
You mean in the `plugin.CreateInRegistry(...` code or in the `CreateInRegistry()` fixture?

In the `plugin.CreateInRegistry(...`, that wouldn't be correct, because it would overwrite other options in the config.

If you meant in the `CreateInRegistry()` fixture; yes, that would work, but would be roughly the same as the change is now.

My train of thought for my first comment was that (I think) `TestPluginInstallArgs` should be able to work with a "default" / "empty" config; from that perspective it would be "cleaner" if we didn't modify the fixture, but adjusted the test to handle that case.

Reply:
I think the embedding pointer to ```PluginConfig``` should be initialized  when cfg ```Config``` be declared.

if check and initial ```cfg.PluginConfig``` in ```CreateOpt``` function, other CreateOpt function (if have) should do so.